sklearn/preprocessing/tests/test_target_encoder.py

```diff
@@ -0,0 +1,304 @@
+    )
+    if target_type == "binary-str":
+        y_input = np.array(["a", "b", "a", "a", "b", "b", "a", "b"])
+        y_int = LabelEncoder().fit_transform(y_input)
+    elif target_type == "binary-ints":
+        y_input = np.array([3, 4, 3, 3, 3, 4, 4, 4])
+        y_int = LabelEncoder().fit_transform(y_input)
+    else:
+        y_input = np.array([3.0, 5.1, 2.4, 3.5, 4.1, 5.5, 10.3, 7.3], dtype=np.float32)
+        y_int = y_input
+    y_mean = np.mean(y_int)
+    categories = [[0, 1, 2], [0, 1]]
+
+    X_test = np.array(
+        [
+            [0, 1],
+            [3, 0],  # 3 is unknown
+            [1, 10],  # 10 is unknown
+        ],
+        dtype=np.int64,
+    )
+
+    if to_pandas:
+        pd = pytest.importorskip("pandas")
+        # convert second feature to a object
+        X_obj = np.array(["cat", "dog"], dtype=object)[X_int[:, 1]]
+        X_input = pd.DataFrame(
+            {"feat0": X_int[:, 0], "feat1": X_obj}, columns=["feat0", "feat1"]
+        )
+        X_test = pd.DataFrame({"feat0": X_test[:, 0], "feat1": ["dog", "cat", "snake"]})
+    else:
+        X_input = X_int
+
+    cv_splits = [([0, 2, 4, 6], [1, 3, 5, 7]), ([1, 3, 5, 7], [0, 2, 4, 6])]
+
+    # manually compute encoding for fit_transform
+    expected_X_fit_transform = np.empty_like(X_int, dtype=np.float64)
+    for f_idx, cats in enumerate(categories):
+        for train_idx, test_idx in cv_splits:
+            X_, y_ = X_int[train_idx, f_idx], y_int[train_idx]
+            current_encoding = _encode_target(X_, y_, len(cats), smooth)
+            expected_X_fit_transform[test_idx, f_idx] = current_encoding[
+                X_int[test_idx, f_idx]
+            ]
+
+    # manually compute encoding for transform
+    expected_encodings = []
+    for f_idx, cats in enumerate(categories):
+        current_encoding = _encode_target(X_int[:, f_idx], y_int, len(cats), smooth)
+        expected_encodings.append(current_encoding)
+
+    expected_X_test_transform = np.array(
+        [
+            [expected_encodings[0][0], expected_encodings[1][1]],
+            [y_mean, expected_encodings[1][0]],
+            [expected_encodings[0][1], y_mean],
+        ],
+        dtype=np.float64,
+    )
+
+    enc = TargetEncoder(smooth=smooth, cv=cv_splits)
+    X_fit_transform = enc.fit_transform(X_input, y_input)
+    assert_allclose(X_fit_transform, expected_X_fit_transform)
+
+    assert len(enc.encodings_) == 2
+    for i in range(2):
+        assert_allclose(enc.encodings_[i], expected_encodings[i])
+
+    X_test_transform = enc.transform(X_test)
+    assert_allclose(X_test_transform, expected_X_test_transform)
+
+
+@pytest.mark.parametrize(
+    "y",
+    [
+        np.array([3.4] * 20),
+        np.array([0] * 20),
+    ],
+    ids=["continuous", "binary"],
+)
+@pytest.mark.parametrize("smooth", ["auto", 4])
+def test_constant_target_and_feature(y, smooth):
+    """Check edge case where feature and target is constant."""
```

Comment:
what happens in the string case for classification? the single class is the 0th class I assume? is it worth testing?

Reply:
Yea, it will become 0. I think it's worth checking.